server/src/main/java/org/apache/druid/segment/realtime/appenderator/UnifiedIndexerAppenderatorsManager.java

```diff
@@ -196,4 +236,288 @@ public boolean shouldTaskMakeNodeAnnouncements()
   {
     return false;
   }
+
+  @Override
+  public void shutdown()
+  {
+    if (mergeExecutor != null) {
+      mergeExecutor.shutdownNow();
+    }
+  }
+
+  private AppenderatorConfig rewriteAppenderatorConfigMemoryLimits(AppenderatorConfig baseConfig)
+  {
+    long perWorkerLimit = workerConfig.getGlobalIngestionHeapLimitBytes() / workerConfig.getCapacity();
+    return new MemoryParameterOverridingAppenderatorConfig(baseConfig, perWorkerLimit);
+  }
+
+  /**
+   * This is a wrapper around AppenderatorConfig that overrides the `maxRowsInMemory` and `maxBytesInMemory`
+   * parameters. Row-based limits are disabled by setting `maxRowsInMemory` to an essentially unlimited value.
+   * `maxBytesInMemory` is overridden with the provided value.
```

Comment:
This javadoc could maybe a little more clearly state that this means that these will override any value the user specifies on the tasks. Also could these by chance be javadoc links to the configs instead of using '`'?

Reply:
I updated the config references and added a note about how this ignores user-specified values